packages/password-input/src/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useControlledValue, useIdAllocator } from '@leafygreen-ui/hooks';
+import LeafyGreenProvider, {
+  useDarkMode,
+} from '@leafygreen-ui/leafygreen-provider';
+import { Label } from '@leafygreen-ui/typography';
+
+import { InputIcon } from '../InputIcon';
+import { StateNotifications } from '../StateNotifications';
+import { TogglePassword } from '../TogglePassword';
+
+import {
+  inputBaseStyles,
+  inputBaseThemeStyles,
+  inputDisabledBaseStyles,
+  inputDisabledThemeStyles,
+  inputIconSizeStyles,
+  inputSizeStyles,
+  inputThemeStyles,
+  inputWrapperStyles,
+  labelBaseStyles,
+  labelLargeOverrideStyles,
+} from './PasswordInput.styles';
+import {
+  type NotificationProps,
+  PasswordInputProps,
+  SizeVariant,
+  States,
+} from './PasswordInput.types';
+import { getStateFromArray } from './utils';
+
+export const PasswordInput = React.forwardRef<
+  HTMLInputElement,
+  PasswordInputProps
+>(
+  (
+    {
+      value: valueProp,
+      onChange: onChangeProp,
+      darkMode: darkModeProp,
+      id: idProp,
+      'aria-describedby': ariaDescribedbyProp,
+      'aria-labelledby': ariaLabelledbyProp,
+      sizeVariant = SizeVariant.Default,
+      stateNotifications = [],
+      disabled = false,
+      autoComplete = 'new-password',
+      className,
+      label,
+      ...rest
+    }: PasswordInputProps,
+    forwardedRef,
+  ) => {
+    const [showPassword, setShowPassword] = useState<boolean>(false);
+    const prefix = 'lg-passwordinput';
+    const inputId = useIdAllocator({ prefix, id: idProp });
+    const descriptionId = useIdAllocator({
+      prefix,
+      id: ariaDescribedbyProp,
+    });
+    const labelId = useIdAllocator({
+      prefix,
+      id: ariaLabelledbyProp,
+    });
+    const { theme, darkMode } = useDarkMode(darkModeProp);
+    const { value, handleChange } = useControlledValue(valueProp, onChangeProp);
+
+    if (!label && !ariaLabelledbyProp && !rest['aria-label']) {
+      console.warn(
+        'For screen-reader accessibility, label, aria-labelledby, or aria-label must be provided',
+      );
+    }
+
+    // If disabled then hide password
+    useEffect(() => {
+      if (disabled) setShowPassword(false);
+    }, [disabled]);
```

Comment:
Why is this default behavior? I could imagine that an input for ENV variables could be disabled but visible until made explicitly visible

Reply:
Hm i didn't consider that. I wonder in that case if the toggle button should never be disabled so that visibility can be toggled when the input is disabled.

Reply:
@brookescarlett

Reply:
Hmm that's a good point. For readonly passwords, I think we should be able to toggle visibility

Reply:
oh yeah, that just clicked -- if its readonly they need something to read lol

Reply:
spoke with Dave, i'll keep the button interactive when the input is disabled.